src/options.ts

```diff
@@ -205,6 +219,42 @@ class SenderOptions {
     }
   }
 
+  /**
+   * Resolves the protocol version, if it is set to 'auto'. <br>
+   * If TCP transport is used, the protocol version will default to 1.
+   * In case of HTTP transport the /settings endpoint of the database is used to find the protocol versions
+   * supported by the server, and the highest will be selected.
+   * @param options SenderOptions instance needs resolving protocol version
+   */
+  static async resolveAuto(options: SenderOptions) {
+    parseProtocolVersion(options);
+    if (options.protocol_version !== PROTOCOL_VERSION_AUTO) {
+      return options;
+    }
+
+    const url = `${options.protocol}://${options.host}:${options.port}/settings`;
+    const settings: {
+      config: { LINE_PROTO_SUPPORT_VERSION: number[] };
+    } = await fetchJson(url);
```

Comment:
We should configure the connection/read timeouts here to match the config ones.

Reply:
fixed in https://github.com/questdb/nodejs-questdb-client/pull/49/commits/812d5b8c7aa3b3062a6fe11359cb87b0f43098ed

Reply:
did not implement the retry options, only a simple timeout.